extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java

```diff
@@ -495,6 +496,12 @@ public Key resolveKey(JsonWebSignature jws, List<JsonWebStructure> nestingContex
                 key = chainResolverFallback.resolveKey(jws, nestingContext);
             }
 
+            if (key == null && oidcConfig.jwks.tryAll && kid == null && thumbprint == null) {
```

Comment:
Can you please move it just above the chainResolverFallback block above ?

Reply:
Also, can you give me a favor and add the same block [here](https://github.com/quarkusio/quarkus/blob/main/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DynamicVerificationKeyResolver.java#L117). I'll look into getting rid of the duplication a bit later, it has been on my map.
I suppose you can make the brute force check a method  on JsonWebKeySet itself for now... thanks

Reply:
I moved the brute force part into `JsonWebKeySet` and added the code as a comment into `Dynamic VerificationKeyResolver` as there is nothing to verify the signature for, so the method cannot be called there.